pkg/apis/proto/sideinput/v1/sideinput.proto

```diff
@@ -0,0 +1,45 @@
+syntax = "proto3";
+
+option go_package = "github.com/numaproj/numaflow-go/pkg/apis/proto/sideinput/v1";
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/empty.proto";
+
+package sideinput.v1;
+
+// UserDefinedSideInput is used to propagate changes in the values of the provided side-inputs
+// which allows access to slow updated data or configuration without needing to retrieve
+// it during each message processing. Through this service the vertex should be able to invoke retrieval request
+// for a single side-input, which in turn should check for updates and return its latest value.
+service UserDefinedSideInput {
+  // RetrieveSideInput applies a function to retrieve the latest side-input values.
+  rpc RetrieveSideInput(google.protobuf.Empty) returns (SideInputResponse);
+
+  // IsReady is the heartbeat endpoint for gRPC.
+  rpc IsReady(google.protobuf.Empty) returns (ReadyResponse);
+}
+
+message EventTime {
+  // event_time is the time associated with side input value to indicate when was the
+  // value updated
+  google.protobuf.Timestamp event_time = 1;
+  // future we can add an enum to represent whether the event time is
+  // based on INGEST_TIME or EVENT_TIME or PROCESSING_TIME
+}
+
+/**
+ * SideInputResponse represents a response to a given side input retrieval request.
+ */
+message SideInputResponse {
+  // value is the side-input payload
+  bytes value = 1;
+  // event_time represents the timestamp at which value is updated
+  EventTime event_time = 2;
```

Comment:
Do we need this?

The call to `RetrieveSideInput` is supposed to be called by the manager sidecar, I would assume for most of the time, there's no way for the developer to determine the time it got updated, and current time would be used for it.

I guess it is introduced for the purpose of determine if there's any change for the data? I think we should use different approaches for it if it's for that.

Reply:
+1, we should remove this as it might cause confusion.